# Patch to set libassimp.dylib's install_name on MacOSX to the final
# installed path of the library. This prevents runtime linking problems
# for apps that link against this library.
# See
# https://github.com/assimp/assimp/issues/166
# https://gist.github.com/dankegel/7125867

--- assimp--3.0.1270-source-only/code/CMakeLists.txt.xyz	2013-10-22 16:37:58.000000000 -0700
+++ assimp--3.0.1270-source-only/code/CMakeLists.txt	2013-10-22 16:40:54.000000000 -0700
@@ -663,6 +663,11 @@
     OUTPUT_NAME assimp${ASSIMP_LIBRARY_SUFFIX}
     CLEAN_DIRECT_OUTPUT 1
 )
+if (APPLE)
+  SET_TARGET_PROPERTIES( assimp PROPERTIES
+    INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}"
+  )
+endif()
 # Build against external unzip, or add ../contrib/unzip so
 # assimp can #include "unzip.h"
 if (UNZIP_FOUND)
